tools/pcap_check.py

```diff
@@ -0,0 +1,39 @@
+#!/usr/bin/python3
+import nasparse
+from scapy.utils import RawPcapNgReader
+import sys
+
+TYPE_LTE_NAS = 0x12
+UDP_LEN = 28
+
+def process_pcap(pcap_path):
+    print('Opening {}...'.format(pcap_path))
+
+    count = 0
+    for (pkt_data, pkt_metadata,) in RawPcapNgReader(pcap_path):
+        count += 1
+        gsmtap_len = pkt_data[UDP_LEN+1]  * 4 # gsmtap header length is stored in the 2nd byte of GSMTAP as a number of 32 bit words
+        header_end = gsmtap_len + UDP_LEN #length of UDP/IP header plus GSMTAP header
+        gsmtap_end_idx = (len(pkt_data) - header_end) * -1
+        
+        gsmtap_hdr = pkt_data[UDP_LEN:gsmtap_end_idx]
```

Comment:
for some reason this took me forever to wrap my head around lol. would `gsmtap_hdr = pkt_data[UDP_LEN:header_end]` work?

Reply:
ah good call, I for some reason forgot you could do a fixed end index for a slice.